app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/TopographyAsyncTest.java

```diff
@@ -0,0 +1,158 @@
+package com.github.ravifrancesco.softwaredevproject;
+
+import android.content.Context;
+
+import androidx.core.util.Pair;
+import androidx.test.core.app.ApplicationProvider;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.osmdroid.util.GeoPoint;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+public class TopographyAsyncTest {
+
+    private static Pair<int[][], Double> topographyPair;
+    private static UserPoint userPoint;
+    private ElevationMapAsync elevationMapAsync;
+
+    /**
+     * Download the topography map
+     * @throws InterruptedException if any thread has interrupted the current thread
+     */
+    @Before
+    public void setup() throws InterruptedException {
+        Context mContext = ApplicationProvider.getApplicationContext();
+
+        userPoint = new UserPoint(mContext);
+        userPoint.setLocation(GPSTracker.DEFAULT_LAT, GPSTracker.DEFAULT_LON,GPSTracker.DEFAULT_ALT, GPSTracker.DEFAULT_ACC);
+        new DownloadTopographyTask(){
+            @Override
+            public void onResponseReceived(Pair<int[][], Double> topography) {
+                super.onResponseReceived(topography);
+                topographyPair = topography;
+            }
+        }.execute(userPoint);
+
+        //Wait for the map to be downloaded
+        int counter=0;
+        while(topographyPair==null && counter<20){
+            Thread.sleep(1000);
+            counter++;
+        }
+        counter=0;
+        Assert.assertNotNull(topographyPair);
+        while(topographyPair.first==null && counter<20){
```

Comment:
What is this while loop for?

Reply:
When testing, I need to wait for the map to be downloaded. If the map is not downloaded, topographyPair.first (which contains the elevation map) is null creating NullPointerExceptions in the other tests.